test/encryption/encryption.test.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const assert = require('assert');
+const mdb = require('mongodb');
+const isBsonType = require('../../lib/helpers/isBsonType');
+
+const LOCAL_KEY = Buffer.from('Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk', 'base64');
+
+describe('environmental variables', () => {
+  it('MONGOOSE_TEST_URI is set', async function() {
+    const uri = process.env.MONGOOSE_TEST_URI;
+    assert.ok(uri);
+  });
+
+  it('CRYPT_SHARED_LIB_PATH is set', async function() {
+    const shared_library_path = process.env.CRYPT_SHARED_LIB_PATH;
+    assert.ok(shared_library_path);
+  });
+});
+
+describe('basic integration', () => {
+  let keyVaultClient;
+  let dataKey;
+  let encryptedClient;
+  let dummyClient;
+
+  beforeEach(async function() {
+    keyVaultClient = new mdb.MongoClient(process.env.MONGOOSE_TEST_URI);
+    await keyVaultClient.connect();
+    await keyVaultClient.db('keyvault').collection('datakeys');
+    const clientEncryption = new mdb.ClientEncryption(keyVaultClient, {
+      keyVaultNamespace: 'keyvault.datakeys',
+      kmsProviders: { local: { key: LOCAL_KEY } }
+    });
+    dataKey = await clientEncryption.createDataKey('local');
+
+    encryptedClient = new mdb.MongoClient(
+      process.env.MONGOOSE_TEST_URI,
+      {
+        autoEncryption: {
+          keyVaultNamespace: 'keyvault.datakeys',
+          kmsProviders: { local: { key: LOCAL_KEY } },
+          schemaMap: {
+            'db.coll': {
+              bsonType: 'object',
+              encryptMetadata: {
+                keyId: [new mdb.UUID(dataKey)]
```

Comment:
I think the return type of createDataKey is `UUID`
```suggestion
                keyId: [dataKey]
```

Reply:
Seems like I was mistaken during our original conversation - this code change works fine - i changed it in the `properties` section as well.